cmd/docker/docker.go

```diff
@@ -44,7 +52,23 @@ func newDockerCommand(dockerCli *command.DockerCli) *cli.TopLevelCommand {
 			return fmt.Errorf("docker: '%s' is not a docker command.\nSee 'docker --help'", args[0])
 		},
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			return isSupported(cmd, dockerCli)
+			if err := isSupported(cmd, dockerCli); err != nil {
+				return err
+			}
+
+			name := cmd.Name()
+			for p := cmd.Parent(); p != nil && p != cmd.Root(); p = p.Parent() {
+				name = p.Name() + " " + name
+			}
+
+			ctx, _ := otel.Tracer("").Start(cmd.Context(), name)
```

Comment:
Just discussing with @milas that we don't have a good convention yet for passing the command name; compose currently uses this;

```
	ctx, cmdSpan := tracing.Tracer.Start(
		ctx,
		"cli/"+strings.Join(commandName(cmd), "-"),
	)
	cmdSpan.SetAttributes(attribute.StringSlice("cli.args", args))
	cmdSpan.SetAttributes(attribute.StringSlice("cli.flags", getFlags(cmd.Flags())))
```


that's what Compose is doing so it's slightly different, e.g. `docker compose alpha publish` -> `cli/alpha-publish`

Reply:
We can keep this as-is for now, and look at improving later; I could add the flags and args if we think it's useful to add.

Reply:
https://opentelemetry.io/docs/specs/semconv/attributes-registry/process/

Reply:
> https://opentelemetry.io/docs/specs/semconv/attributes-registry/process/

OH! That looks great, and from the looks of it that pretty much describes the things we'd want to include (and we wouldn't have to come up with our own conventions, which is a big 💯)

@milas @jsternberg You'll probably be interested in this (if you didn't know about these).

From the looks of it; most of these we could set using a generic helper (if none exists for it). We _MAY_ have to consider if we should sanitise arguments though (i.e. do we want `--env MYSQL_ROOT_PASSWORD=<pass>` (or `--build-arg`) to end up in traces? 😬)

If we decide to sanitise, perhaps we need an option to opt-out of sanitising (for those that _DO_ want to include all of that in their traces).

Let me add a screenshot to save others from having to follow the link, and to add context to my comment;

<img width="1022" alt="Screenshot 2024-01-05 at 12 28 37" src="https://github.com/docker/cli/assets/1804568/d56fda60-4f05-4a2b-bfb1-1be532f100a8">